src/module/actor/sheet/base.ts

```diff
@@ -219,6 +231,10 @@ abstract class ActorSheetPF2e<TActor extends ActorPF2e> extends ActorSheet<TActo
     }
 
     protected prepareInventoryItem(item: PhysicalItemPF2e): InventoryItem {
+        const items = [...iterateAllItems(this.actor)].filter((i) => i.isOfType("physical"));
+        this.#inventorySearchEngine.removeAll();
+        this.#inventorySearchEngine.addAll(items.map((i) => R.pick(i, ["id", "name"])));
```

Comment:
This should be in prepareInventory(), not prepareInventoryItem(). prepareInventoryItem() is called per item.